File: src/main/java/com/sup/dev/java_pc/sql/SqlQuery.java
package com.sup.dev.java_pc.sql;

public abstract class SqlQuery extends Sql {

    private String query;

    protected abstract String createQuery();

    //
    //  Getters
    //

    @Override
    public String toString() {
        return getQuery();
    }

    public String getQuery() {
        if(query == null)
            query = createQuery();
        return query;
    }

}

File: src/main/java/com/sup/dev/java_pc/sql/SqlQuery.kt
package com.sup.dev.java_pc.sql


abstract class SqlQuery{

    private var query: String? = null

    protected abstract fun createQuery(): String

    //
    //  Getters
    //

    override fun toString(): String {
        return getQuery()!!
    }

    fun getQuery(): String? {
        if (query == null)
            query = createQuery()
        return query
    }

}
